### DISCUSSION TITLE
How to do logic synthesis only optimize on selected modules and don't touch other modules ?

### DISCUSSION THREAD

**Discussion by chestercc1997** [YosysHQ/yosys#4823] [Q&A]
If I have 3 modules, I want to perform optimization using ABC only on module 1, while keeping module 2 and module 3 unoptimized. After optimization, I would like to generate AIG/Blif. Are there any scripts that support this functionality?

**Comment by wjrforcyber** [comment]
If you are using command `abc`, first check the manual [here](https://yosys.readthedocs.io/_/downloads/en/latest/pdf/), or just type `help abc` in yosys.

>If I have 3 modules, I want to perform optimization using ABC only on module 1, while keeping module 2 and module 3 unoptimized. 

This is a temporary check process, you could modify the code to achieve the result.
Read file https://github.com/YosysHQ/yosys/blob/main/passes/techmap/abc.cc , line:
https://github.com/YosysHQ/yosys/blob/8acc77c1e0e7c67ab174875556ce6743faf9efae/passes/techmap/abc.cc#L2045
is iterating modules, select the module you would like to optimise or just set breakpoint here, with option:
https://github.com/YosysHQ/yosys/blob/8acc77c1e0e7c67ab174875556ce6743faf9efae/passes/techmap/abc.cc#L1619-L1622

and option:
https://github.com/YosysHQ/yosys/blob/8acc77c1e0e7c67ab174875556ce6743faf9efae/passes/techmap/abc.cc#L1623-L1626
The result blif will be saved at:
https://github.com/YosysHQ/yosys/blob/8acc77c1e0e7c67ab174875556ce6743faf9efae/passes/techmap/abc.cc#L865

If you do not enable `-nocleanup`, the temporary file will be removed at the end:
https://github.com/YosysHQ/yosys/blob/8acc77c1e0e7c67ab174875556ce6743faf9efae/passes/techmap/abc.cc#L1445-L1449
